godot-core/src/macros.rs

```diff
@@ -484,6 +484,44 @@ macro_rules! gdext_virtual_method_callback {
         )
     };
 
+    // immutable
```

Comment:
This macro is getting out of hand, we really need to translate it to proc-macros at some point 😉 

(Don't worry, not as part of this PR)

Reply:
Completely agree. I was hoping to find some alternative, but `macro_rules!` is a bit too limited to match what we want here without brute forcing. Maybe a proc macro? We can chat on Discord; I'm interested in a good solution to this problem as well.